table_test.go

```diff
@@ -48,6 +48,60 @@ func TestBucket(t *testing.T) {
 	}
 }
 
+func TestGenRandPeerID(t *testing.T) {
+	nBuckets := 16
+	local := test.RandPeerIDFatal(t)
+	m := pstore.NewMetrics()
+	rt := NewRoutingTable(1, ConvertPeerID(local), time.Hour, m)
+
+	// create nBuckets
+	for i := 0; i < nBuckets; i++ {
+		for {
+			if p := test.RandPeerIDFatal(t); CommonPrefixLen(ConvertPeerID(local), ConvertPeerID(p)) == i {
+				rt.Update(p)
+				break
+			}
+		}
+	}
+
+	// test bucket for peer
+	peers := rt.ListPeers()
+	for _, p := range peers {
+		b := rt.BucketForPeer(p)
+		if !b.Has(p) {
+			t.Fatalf("bucket should have peers %s", p.String())
+		}
+	}
+
+	// test generate rand peer ID
+	for bucketID := 0; bucketID < nBuckets; bucketID++ {
```

Comment:
Let's test this with with more buckets (40?) to check for overflow issues.

Reply:
@Stebalien  Generating 40 buckets takes a lot of time as a new bucket is created ONLY IF the last bucket is full. This slows down the tests. So testing with 21 for now as it's still an overflow & hits all the code-paths.

Reply:
It _shouldn't_ be slow... We shouldn't have to generate 40 buckets, just ask for peer IDs in buckets 1-40 (where everything after 16 would just be random IDs.

But this should be good enough.

Reply:
@Stebalien The problem is that: 

1) We don't add a peer to a bucket other than the last bucket if it is already full (assume the latency stuff dosen't picture here)
2) We split the last bucket & add a peer to it if & only if it 'belongs' to the last bucket.

So, let's say you have 22 buckets all of which are full. Now, to create a 23rd bucket, you have to 
generate an ID with a CPL of atleast 22. That can be not be done deterministically by our GenRandPeerID(> 16). So the only way to is to keep trying till we get lucky.

Reply:
Yes. But really, anything over 16 is plenty.